app/controllers/user_projects_controller.rb

```diff
@@ -275,6 +276,24 @@ def check_delete_permisisons
     end
   end
 
+  def generate_notification(user, target_user, project)
```

Comment:
I think that this method is duplicated in 2 places now here and in settings_controller. 
Can you refactor the method that will work in both cases and put it in a helper file?